backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/objcexport/ObjCExportCodeGenerator.kt

```diff
@@ -1563,34 +1563,22 @@ internal fun ObjCExportCodeGenerator.getEncoding(methodBridge: MethodBridge): St
         }
     }
 
-    val targetFamily = context.config.target.family
-    val returnTypeEncoding = methodBridge.returnBridge.getObjCEncoding(targetFamily)
+    val returnTypeEncoding = methodBridge.returnBridge.getObjCEncoding(context)
 
     val paramSize = paramOffset
     return "$returnTypeEncoding$paramSize$params"
 }
 
-// https://developer.apple.com/documentation/objectivec/nsuinteger?language=objc
-// `typedef unsigned long NSUInteger` on iOS, macOS, tvOS.
-// `typedef unsigned int NSInteger` on watchOS.
-private val Family.nsUIntegerEncoding: String get() = when (this) {
-    Family.OSX,
-    Family.IOS,
-    Family.TVOS -> "L"
-    Family.WATCHOS -> "I"
-    else -> error("Unexpected target platform: $this")
-}
-
-private fun MethodBridge.ReturnValue.getObjCEncoding(targetFamily: Family): String = when (this) {
+private fun MethodBridge.ReturnValue.getObjCEncoding(context: Context): String = when (this) {
     MethodBridge.ReturnValue.Suspend,
     MethodBridge.ReturnValue.Void -> "v"
-    MethodBridge.ReturnValue.HashCode -> targetFamily.nsUIntegerEncoding
+    MethodBridge.ReturnValue.HashCode -> if (context.is64BitNSInteger()) "Q" else "I"
```

Comment:
IIUC, It should be "L" for ios_arm32:
* `NSUInteger` is an alias for `unsigned long` on iOS.
* `unsigned long` is mapped to "L" on 32-bit platforms.
Am I missing something?

Reply:
See [comment in the issue](https://youtrack.jetbrains.com/issue/KT-44083#focus=Comments-27-4644443.0-0) with details of the `clang` output. It shows that it encodes to `I` on 32-bit iOS.